### ISSUE TITLE
Rework integration tests

### ISSUE THREAD

**Issue by fkuatspyro** [Spyro-Soft/scargo#381]
Currently integration tests ([it_scargo_commands_flow](https://github.com/Spyro-Soft/scargo/blob/release_1.8.0/tests/it/it_scargo_commands_flow.py)) are written using pytest.mark.order to ensure correct order of commands. This pytest is being misused as the tests rely on runs of previous tests.

There are multiple problems with current implementation.
1. **Some of the "testcases" aren't even tests**
_test_precondition_add_new_profile_ -> just adds profile to toml file, does not test anything
2. **Test case is NOT test setup**
Fixtures are meant to be used as a setup, not previous test case. The above mentioned _test_precondition_add_new_profile_ is a precondition to the following _test_cli_update_after_adding_new_profile_. How is one supposed to run this single test? pytest.mark.order is not a fixure and it doesn't require that after argument is run before the test is executed.
Running `pytest tests/it/it_scargo_commands_flow.py::TestBinProjectFlow::test_cli_update_after_adding_new_profile[new_bin_project_x86]`
will fail with `FileNotFoundError: [Errno 2] No such file or directory: 'None/new_bin_project_x86`. In other words, there is no setup for this test and project directory doesn't exists as the project itself is never created.
3. **It's really hard to follow what is going on.**
The file has >800 lines of code. Trying to figure out what is happening and what is being run after what is pretty much impossible, 
4. **Some of the test cases should be unit tests**
_test_cli_update_after_adding_new_profile_ test case is an example of easily mockable test that should be unit test. There is no way that all of this setup is needed to check that scargo update call writes specific C and CXX flags.

**Comment by aaksenczuk** [comment]
fix and will be delivered with 2.0.2
